include/rocksdb/file_system.h

```diff
@@ -777,10 +790,14 @@ class FSWritableFile {
     return IOStatus::NotSupported("PositionedAppend");
   }
 
-  // EXPERIMENTAL / CURRENTLY UNUSED
   // PositionedAppend data with verification information.
   // Note that this API change is experimental and it might be changed in
-  // the future. Currently, RocksDB does not use this API.
+  // the future. Currently, RocksDB only generates crc32c based checksum for
+  // the file writes when the option is set.
+  // Expected behavior: if the supported checksum verification type in this
+  // FSWritableFile does not match the handoff_checksum_type in FileOptions,
```

Comment:
Do you mean FileOptions passed in the constructor? Why do we pass the checksum type twice in the constructor and only use it if they agree?

Reply:
Sorry, I mixed up the FileOptions and FSWritableFile constructors.

"the supported checksum verification type in this FSWritableFile" - I think this confused me a bit because it sounds like the FSWritableFile has a field for the checksum type. What I think you mean now is (rephrased) "the checksum verification type supported by this FSWritableFile". Suggested edit:

    // Expected behavior: if the handoff_checksum_type in FileOptions is not supported by
    // this FSWritableFile, the information in DataVerificationInfo should be ignored (i.e. does
    // not perform checksum verification)

I suggest we also consider changing "should" to "can," so that implementations can choose to "fail safe" in the case of a configuration error rather than "quietly skip verification." But it's up to you, and I'm not blocking approval on it.

Reply:
Thanks for the clarification. I will use the suggested statement in the comments. Also, using "can" makes it more flexible such that the developer of FSWritableFile can decide the behavior, like "fail" the writes if they want.